executor/batch_point_get_test.go

```diff
@@ -162,3 +166,138 @@ func (s *testBatchPointGetSuite) TestBatchPointGetUnsignedHandleWithSort(c *C) {
+
+		tk1.MustExec("use test")
+		tk2.MustExec("use test")
+		tk1.MustExec("set session tidb_enable_clustered_index = 0")
+
+		tk1.MustExec(fmt.Sprintf("drop table if exists %s", tableName))
+		tk1.MustExec(fmt.Sprintf("create table %s(id int, v int, k int, %s key0(id, v))", tableName, key))
+		tk1.MustExec(fmt.Sprintf("insert into %s values(1, 1, 1), (2, 2, 2)", tableName))
+
+		if rc {
+			tk1.MustExec("set tx_isolation = 'READ-COMMITTED'")
+			tk2.MustExec("set tx_isolation = 'READ-COMMITTED'")
+		}
+
+		tk1.MustExec("begin pessimistic")
+		tk2.MustExec("begin pessimistic")
+
+		// select for update
+		if !rc {
+			// lock exist key only for repeatable read
+			tk1.MustExec(fmt.Sprintf("select * from %s where (id, v) in ((1, 1), (2, 2)) for update", tableName))
+		} else {
+			// read committed will not lock non-exist key
+			tk1.MustExec(fmt.Sprintf("select * from %s where (id, v) in ((1, 1), (2, 2), (3, 3)) for update", tableName))
+		}
+		tk2.MustExec(fmt.Sprintf("insert into %s values(3, 3, 3)", tableName))
+		go func() {
+			tk2.MustExec(fmt.Sprintf("insert into %s values(1, 1, 10)", tableName))
+			doneCh <- struct{}{}
+		}()
+
+		time.Sleep(150 * time.Millisecond)
+		tk1.MustExec(fmt.Sprintf("update %s set v = 2 where id = 1 and v = 1", tableName))
+
+		tk1.MustExec("commit")
+		<-doneCh
+		tk2.MustExec("commit")
+		tk1.MustQuery(fmt.Sprintf("select * from %s", tableName)).Check(testkit.Rows(
+			"1 2 1",
+			"2 2 2",
+			"3 3 3",
+			"1 1 10",
+		))
+
+		// update
+		tk1.MustExec("begin pessimistic")
+		tk2.MustExec("begin pessimistic")
+		if !rc {
+			// lock exist key only for repeatable read
+			tk1.MustExec(fmt.Sprintf("update %s set v = v + 1 where (id, v) in ((2, 2), (3, 3))", tableName))
+		} else {
+			// read committed will not lock non-exist key
+			tk1.MustExec(fmt.Sprintf("update %s set v = v + 1 where (id, v) in ((2, 2), (3, 3), (4, 4))", tableName))
+		}
+		tk2.MustExec(fmt.Sprintf("insert into %s values(4, 4, 4)", tableName))
+		go func() {
+			tk2.MustExec(fmt.Sprintf("insert into %s values(3, 3, 30)", tableName))
+			doneCh <- struct{}{}
+		}()
+		time.Sleep(150 * time.Millisecond)
+		tk1.MustExec("commit")
+		<-doneCh
+		tk2.MustExec("commit")
+		tk1.MustQuery(fmt.Sprintf("select * from %s", tableName)).Check(testkit.Rows(
+			"1 2 1",
+			"2 3 2",
+			"3 4 3",
+			"1 1 10",
+			"4 4 4",
+			"3 3 30",
+		))
+
+		// delete
+		tk1.MustExec("begin pessimistic")
+		tk2.MustExec("begin pessimistic")
+		if !rc {
+			// lock exist key only for repeatable read
+			tk1.MustExec(fmt.Sprintf("delete from %s where (id, v) in ((3, 4), (4, 4))", tableName))
+		} else {
+			// read committed will not lock non-exist key
+			tk1.MustExec(fmt.Sprintf("delete from %s where (id, v) in ((3, 4), (4, 4), (5, 5))", tableName))
+		}
+		tk2.MustExec(fmt.Sprintf("insert into %s values(5, 5, 5)", tableName))
+		go func() {
+			tk2.MustExec(fmt.Sprintf("insert into %s values(4, 4,40)", tableName))
```

Comment:
We could use the lock error returned by the `select * from t where (id, v) in (4, 4,  40) for update nowait` to verify the lock does exist. 
`MustExec` in anohter goroutine seems will not cause test failure if failed?

Reply:
`select for update nowait` statement will lead to fail due to lock on row key, I use an err channel to collect the exec error and assert it in the test function.